merge_fastqs.py

```diff
@@ -0,0 +1,85 @@
+# Modified version of https://github.com/SciLifeLab/standalone_scripts/blob/master/merge_and_rename_NGI_fastq_files.py
+# to fit the NGI Uppsala standard
+
+import re
+import os
+import sys
+import shutil
+import argparse
+import collections
+
+def merge_files(input_dir, dest_dir):
+
+    #Gather all fastq files in inputdir and its subdirs
+    fastq_files=[]
+    for subdir, dirs, files in os.walk(input_dir):
+        for fastq in  files:
+            if fastq.endswith('.fastq.gz'):
+                fastq_files.append(os.path.join(subdir, fastq))
+
+    #Match NGI sample number from flowcell
+    sample_pattern=re.compile("^(.+)_S[0-9]+_L00[1-8]_R([1-2])_")
+    #Remove files that already have the right name (i.e have been merged already)
+    matches=[]
+    for fastq_file in fastq_files:
+        try:
+            match=sample_pattern.search(os.path.basename(fastq_file)).group(1)
+            if match:
+                matches.append(fastq_file)
+        except AttributeError:
+            continue
+    fastq_files=matches
+
+    while fastq_files:
+        tomerge=[]
+
+        #grab one sample to work on
+        first=fastq_files[0]
+        fq_bn=os.path.basename(first)
+        sample_name=''.join((sample_pattern.match(fq_bn).group(1),'_'))
+        fastq_files_read1=[]
+        fastq_files_read2=[]
+
+        for fq in fastq_files:
+            if sample_name in os.path.basename(fq) and "_R1_" in os.path.basename(fq):
```

Comment:
note that there is a bug here! 

e.g. "sample_1" is a substring of "sample_11_R1.fastq.gz" but you don't want to merge those

Reply:
New script included in new commit